linode/manager
# M3-5585: Add Story for Navigation

## Description

This PR adds stories for all things menu related (ie. `TopMenu`, `SideMenu`, and `UserMenu`) along with mock profile data.

**Update:** Moved the controls for the `SideMenu` to the `ArgsTable` and also imported the Storybook Controls add-on so the props can also be adjusted under the Canvas tab. This was implemented to account for the `SideMenu` leaking out of the container for viewports less than 960px wide. 

The latest version of the Storybook Controls add-on causes react-router-dom to break so we're using an older version.
[Storybook v6.4 is breaking users using react-router v5](https://github.com/storybookjs/storybook/issues/16837)

## How to test

- Run `yarn storybook` and go to [http://localhost:6006/?path=/docs/components-navigation-top-menu--top-menu](http://localhost:6006/?path=/docs/components-navigation-top-menu--top-menu)
- See the stories for `TopMenu`, `SideMenu`, and `UserMenu`
- I added a conditional for undefined `gravatarURL`s so check for any regressions related to that
